src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -80,8 +78,20 @@ private static object InvokeExpression(Expression expression)
             return Expression.Lambda(expression).Compile().DynamicInvoke();
         }
 
+        private static void CheckArgumentExpressionIsValid(Expression expression)
+        {
+            var visitor = new ArgumentConstraintExpressionVisitor();
+            visitor.Visit(expression);
+            if (visitor.ConstraintExpressions.Any(e => e != expression))
+            {
+                throw new InvalidOperationException(ExceptionMessages.ArgumentConstraintCannotBeNestedInArgument);

```

Comment:
That was my initial intention, but then I realized that getting a sensible representation of the expression was far from trivial; we don't have access to the original source code, as in the analyzer. And anyway, should we show the expression in C#, VB, something else?